staging/src/k8s.io/kms/encryption/managed.go

```diff
@@ -0,0 +1,264 @@
+package encryption
+
+import (
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"k8s.io/klog/v2"
+)
+
+const (
+	// MaxUsage with 2^21 is a very defensive value. 2^32 is more commonly used.
+	MaxUsage = 2097151
+	// keySize is the key size in bytes
+	keySize = 128 / 8
+	// nonceSize is the size of the nonce. Do not change, without breaking version change.
+	// The nonceSize is a de facto standard.
+	nonceSize = 12
+	// cacheSize is set to 100 keys, which can be used to decrypt up to 200 million encrypted
+	// files or data encrypted within nearly 2 years.
+	cacheSize = 100
+)
+
+var (
+	// ErrKeyExpired means that the expiration time of a key has come and it shouldn't be used any more.
+	ErrKeyExpired = errors.New("key is out of date and shouldn't be used anymore for encryption")
+	// ErrNoCipher means that there is no remote kms given and therefore the keys in use can't be protected.
+	ErrNoCipher = errors.New("no remote encryption service was specified")
+
+	week = time.Hour * 24 * 7
```

Comment:
Should we allow users to make this shorter?

Reply:
Sure, users could fine tune it to their needs. The max would stay at one week?